backend/fms_core/viewsets/full_sample.py

```diff
@@ -0,0 +1,216 @@
+            container_obj = Container.objects.get(pk=full_sample['container'])
+        except Exception as err:
+            error['container'] = err
+
+        #Individual related information
+        individual_obj = None
+        if full_sample['individual']:
+            try:
+                individual_obj = Individual.objects.get(pk=full_sample['individual'])
+            except Exception as err:
+                error['individual'] = err
+
+        #Sample Kind related information
+        try:
+            sample_kind_obj = SampleKind.objects.get(pk=full_sample['sample_kind'])
+        except Exception as err:
+            error['sample_kind'] = err
+
+        biosample_data = dict(
+            collection_site=full_sample['collection_site'],
+            **(dict(individual=individual_obj) if individual_obj is not None else dict()),
+            **(dict(alias=full_sample['alias']) if full_sample['alias'] is not None else dict()),
+        )
+
+        try:
+            biosample = Biosample.objects.create(**biosample_data)
+
+            derived_sample_data = dict(
+                biosample_id=biosample.id,
+                sample_kind=sample_kind_obj,
+                **(dict(tissue_source=full_sample['tissue_source']) if full_sample['tissue_source'] is not None else dict()),
+            )
+            if full_sample['experimental_group']:
+                derived_sample_data['experimental_group'] = json.dumps([
+                    g.strip()
+                    for g in RE_SEPARATOR.split(full_sample['experimental_group'])
+                    if g.strip()
+                ])
+
+            derived_sample = DerivedSample.objects.create(**derived_sample_data)
+
+            sample_data = dict(
+                name=full_sample['name'],
+                volume=full_sample['volume'],
+                creation_date=full_sample['creation_date'],
+                container=container_obj,
+                comment=(full_sample['comment'] or (f"Automatically generated on {datetime.utcnow().isoformat()}Z")),
+                **(dict(coordinates=full_sample['coordinates']) if full_sample['coordinates'] is not None else dict()),
+                **(dict(concentration=full_sample['concentration']) if full_sample['concentration'] is not None else dict()),
+            )
+
+            sample = Sample.objects.create(**sample_data)
+
+            DerivedBySample.objects.create(derived_sample_id=derived_sample.id,
+                                           sample_id=sample.id,
+                                           volume_ratio=1)
+
+        except ValidationError as err:
+            raise ValidationError(err)
+
+        serializer = SampleSerializer(sample)
+        data = serializer.data
+        return Response(data)
+
+    def update(self, request, *args, **kwargs):
+        full_sample = request.data
+
+        #Retrive the sample to update
+        try:
+            sample_to_update = Sample.objects.get(pk=full_sample['id'])
```

Comment:
sample_to_update = Sample(**request.data) ?

Reply:
What?

Reply:
you already have the date and id ... you can just load the object using the data and save it as far as i know, maybe you need to add the tracking info ... but i am not sure you need to go one attribute at a time fetching intermediate objects.